queenbee/config/repositories.py

```diff
@@ -0,0 +1,65 @@
+import os
+from urllib.parse import urlparse
+from pydantic import Field, validator
+from ..base.basemodel import BaseModel
+from ..base.request import make_request, urljoin
+
+class RepositoryReference(BaseModel):
+
+  name: str = Field(
+    ...,
+    description='The name of the repository'
+  )
+
+  path: str = Field(
+    ...,
+    description='The path to the repository'
+  )
+
+  @validator('path')
+  def remote_or_local(cls, v):
+    """Determine whether the path is local or remote (ie: http)"""
+    parsed_path = urlparse(v)
+
+    # If it's on the local machine
+    if parsed_path.scheme == '':
+      v = f'file:{v}'
+
+    return v
+
+  def fetch(self, auth_header: str = '') -> 'RepositoryIndex':
```

Comment:
I'm unfamiliar with this arrow notation. It looks like it's a [type annotation](https://www.python.org/dev/peps/pep-3107/#fundamentals-of-function-annotations)?

Is there a difference between 'returning' a string here and a reference to the actual type? You consistently do strings, and the document says that would have to be interpreted by a third party library, so I'm assuming not. Figured it's still worth learning/confirming here.

Reply:
Yep this is type annotations introduced in [PEP 484](https://www.python.org/dev/peps/pep-0484/).

You normally would us a class directly rather than a string indeed but in this case I use a string because importing the class in question would either lead to recursive imports or because this is a classmethod (and therefore the class isn't fully declared yet within the scope of this type declaration).